### ISSUE TITLE
[Bug] Drawer is missing ARIA role

### ISSUE THREAD

**Issue by Racer159** [defenseunicorns/UnicornUI#231]
**Describe the bug**

Currently the Drawer component is missing an ARIA role and when it is opened it doesn't take focus leaving the screen reader still in the main content without notifying the user.

**To Reproduce**
1. Start a screen reader on your system
2. In Zarf - open a package for deployment
3. Tab over to the "Code" button and press enter
4. Continue to Tab and use arrow keys to navigate
(noticing you are still in the main content and there was no notification that a div appeared)

**Expected behavior**
Focus is taken to the header of the Drawer and the user is told that something happened (ideally reading the header)

**Screenshots**
![image](https://github.com/defenseunicorns/UnicornUI/assets/3977569/f4ea9ad0-2d80-4fb3-979a-e00b8550b392)
![image](https://github.com/defenseunicorns/UnicornUI/assets/3977569/32372780-3003-4a89-8f0d-ea5b12626d67)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, safari] `Firefox`
 - UnicornUI Version [e.g. 22] `49`

**Additional context**
This could lead to confusion for people who are unable to read the screen (in the worst case) and due to focus handling makes it more difficult for people to use with a keyboard (given there is no easy way to get over to the drawer without tabbing through a lot more content)
